firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -131,49 +122,39 @@ bool moveStraight(int motor[], float startCol, float startRow, float endCol, flo
   // Rotate motor some number of steps
   for (i = 0; i < numSteps; i++) 
   {
-    if (digitalRead(X_AXIS_ENDSTOP_SWITCH) == HIGH  ||  
-        digitalRead(Y_AXIS_ENDSTOP_SWITCH) == HIGH)
-      return false;
+    if (digitalRead(X_AXIS_ENDSTOP_SWITCH) == HIGH)
+      return HIT_X_ENDSTOP;  
+    else if (digitalRead(Y_AXIS_ENDSTOP_SWITCH) == HIGH)
```

Comment:
This can just be an `if` statement.

If the first if statement is true, then you will return, (so the else if does nothing). However, if the first if statement is false, you'll check the next condition anyway (so the else if does nothing).

The [linter](https://pylint.org) I use for checking my python code always points this out when I do the same thing lol.

TL;DR, it isn't an error and it has same runtime, but I think for style you should use an `if` instead of an `else if` here.

Reply:
Even though it works both ways, having two if statements makes sense rather than the else statements. It has a similar scheme to checking for base cases in recursion